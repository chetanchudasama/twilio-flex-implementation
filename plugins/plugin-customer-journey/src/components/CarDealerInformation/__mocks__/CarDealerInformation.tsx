import React from "react";
import { Props } from "../CarDealerInformation";

const CarDealerInformation: React.FC<Props> = (props) => <div />;

export default CarDealerInformation;
