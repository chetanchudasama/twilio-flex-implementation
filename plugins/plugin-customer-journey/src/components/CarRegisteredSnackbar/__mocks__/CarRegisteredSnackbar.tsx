import React from "react";
import { CarRegisteredSnackbarProps } from "../CarRegisteredSnackbar.Props";

export const CarRegisteredSnackbar: React.FC<CarRegisteredSnackbarProps> = (
  props
) => {
  return <div className="car-registered-snackbar">CarRegisteredSnackbar</div>;
};
