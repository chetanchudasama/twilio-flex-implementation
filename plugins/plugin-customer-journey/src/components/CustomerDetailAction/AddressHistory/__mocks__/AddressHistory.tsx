import React from "react";
import { AddressHistoryProps } from "../AddressHistory";

const AddressHistory: React.FC<AddressHistoryProps> = (props) => (
  <div className="address-history" />
);

export default AddressHistory;
