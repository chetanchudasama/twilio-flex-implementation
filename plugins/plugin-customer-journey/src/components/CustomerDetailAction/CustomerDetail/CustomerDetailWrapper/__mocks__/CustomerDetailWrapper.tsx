import React from "react";
import { ComponentProps } from "../CustomerDetailWrapper";

const CustomerDetailWrapper: React.FC<ComponentProps> = (props) => (
  <div className="customer-detail-wrapper" />
);

export default CustomerDetailWrapper;
