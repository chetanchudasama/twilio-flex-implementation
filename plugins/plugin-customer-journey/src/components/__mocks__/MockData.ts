import { WorkerAttributes } from "@twilio/flex-ui";
import { namespace } from "../../states";

const workerAttributes: WorkerAttributes = {
  contact_uri: "client:test",
  email: "test@test.com",
  full_name: "test user",
  image_url: "img_url",
  roles: ["admin"],
  agentId: 1,
};

const conferenceSource = {
  source: {
    participants: [
      {
        callSid: "callSid",
        participantType: "customer",
        workerSid: "",
      },
      {
        callSid: "callSid",
        participantType: "worker",
        workerSid: "workerSid",
      },
    ],
  },
};

const conferenceState = new Map();
conferenceState.set("taskSid", conferenceSource);

export const flexManagerMock = jest.fn().mockReturnValue({
  workerClient: {
    source: {},
    sid: "workerSid",
    name: "test user",
    fullName: "test user",
    activityName: "Offline",
    attributes: workerAttributes,
    dateUpdated: new Date(),
    isAvailable: false,
    activityDuration: "",
    dateStatusChanged: new Date(1).toISOString(),
    activity: {
      available: false,
      name: "Offline",
      sid: "OfflineSid",
    },
  },
  store: {
    getState: jest.fn().mockReturnValue({
      flex: {
        worker: {
          attributes: workerAttributes,
        },
        session: {
          ssoTokenPayload: {
            token: "ssoToken",
          },
        },
        conferences: {
          states: conferenceState,
        },
      },
      [namespace]: {
        phoneNumberList: {
          phoneNumberList: [
            {
              friendlyName: "Sales",
              phoneNumber: "+1234567891",
              queueSid: "QUEUE_SID",
            },
          ],
        },
      },
    }),
  },
  strings: {
    DIALPADExternalTransferSearchInputLabel: "Search By Name or PhoneNumber",
    DIALPADExternalTransferHoverOver: "Add",
    DIALPADExternalTransferPopupHeader: "External Transfer",
  },
  serviceConfiguration: {
    outbound_call_flows: {
      default: {
        caller_id: "CALLER_ID",
      },
    },
  },
});
