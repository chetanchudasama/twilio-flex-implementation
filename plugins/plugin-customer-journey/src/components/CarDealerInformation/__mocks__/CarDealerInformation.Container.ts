import CarDealerInformation from "./CarDealerInformation";

export default CarDealerInformation;
