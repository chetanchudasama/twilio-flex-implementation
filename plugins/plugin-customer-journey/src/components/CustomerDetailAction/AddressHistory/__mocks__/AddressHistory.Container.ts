import AddressHistory from "./AddressHistory";

export default AddressHistory;
