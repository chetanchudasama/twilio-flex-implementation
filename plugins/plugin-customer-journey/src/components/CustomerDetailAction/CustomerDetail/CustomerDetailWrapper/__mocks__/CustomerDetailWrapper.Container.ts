import CustomerDetailWrapper from "./CustomerDetailWrapper";

export default CustomerDetailWrapper;
