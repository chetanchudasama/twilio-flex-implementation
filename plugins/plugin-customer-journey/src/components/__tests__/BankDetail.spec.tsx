import React from "react";
import { create } from "react-test-renderer";
import flushPromises from "flush-promises";
import { mount } from "enzyme";
import { act } from "react-dom/test-utils";
import { Manager } from "@twilio/flex-ui";
import { showErrorMessage, showMessage } from "../../Notifications";
import {
  BankDetail,
  ComponentProps,
} from "../CustomerDetailAction/BankDetail/BankDetail";
import { flexManagerMock } from "../__mocks__/MockData";

jest.mock("../../services/application.service");

jest.mock("../../Notifications", () => {
  return {
    showErrorMessage: jest.fn(() => {}),
    showMessage: jest.fn(() => {}),
    CustomNotificationType: { SuccessNotification: "successNotification" },
  };
});

const mockUpdateBaseApplicationDetail = jest.fn();

jest.mock("../../services/application.service", () => ({
  useApplicationService: jest.fn(() => ({
    updateBaseApplicationDetail: mockUpdateBaseApplicationDetail,
  })),
}));

jest.mock("../LoadingComponent/Loading.Container", () => {
  return {
    __esModule: true,
    default: () => {
      return <div>Loading</div>;
    },
  };
});

Manager.getInstance = flexManagerMock;

const props: ComponentProps = {
  applicationId: 1,
  bankDetail: {
    bankName: "Bank of England",
    monthsAtBank: 11,
    nameOnBankAccount: "Edward Gibbs",
    accountNumber: "31510604",
    sortCode: "565777",
    netMonthlyIncome: 30000,
  },
  setCustomerDetails: jest.fn(() => {}),
  backToWizardStepper: jest.fn(() => {}),
};

describe("BankDetail", () => {
  afterEach(() => {
    jest.clearAllMocks();
    jest.clearAllTimers();
  });

  it("renders correctly", async () => {
    const wrapper = create(<BankDetail {...props} />).toJSON();
    await flushPromises();

    expect(wrapper).toMatchSnapshot();
  });

  it("should render all input fields correctly in UI", () => {
    const wrapper = create(<BankDetail {...props} />).root;

    expect(wrapper.findAllByType("input")).toHaveLength(6);
  });

  it("when update detail button is clicked, should update bank detail and show success notification", async () => {
    mockUpdateBaseApplicationDetail.mockImplementationOnce(() =>
      Promise.resolve(undefined)
    );

    await act(async () => {
      const wrapper = mount(<BankDetail {...props} />);
      const input = wrapper
        .find(".monthly-income-container")
        .last()
        .find("input")
        .last();
      input.simulate("change", { target: { value: "1000" } });
      await flushPromises();
      wrapper.update();

      wrapper.find(".update-btn").at(0).simulate("click");
      await flushPromises();
      wrapper.update();

      expect(mockUpdateBaseApplicationDetail).toHaveBeenCalledTimes(1);
      expect(showMessage).toHaveBeenCalledWith(
        "successNotification",
        "Customer bank detail updated successfully!"
      );
      expect(props.setCustomerDetails).toHaveBeenCalled();
      expect(props.backToWizardStepper).toHaveBeenCalled();
    });
  });

  it("should show error message, if getting error while update bank details", async () => {
    mockUpdateBaseApplicationDetail.mockImplementationOnce(() =>
      Promise.reject()
    );

    await act(async () => {
      const wrapper = mount(<BankDetail {...props} />);
      const input = wrapper
        .find(".monthly-income-container")
        .last()
        .find("input")
        .last();
      input.simulate("change", { target: { value: "1000" } });
      await flushPromises();
      wrapper.update();

      wrapper.find(".update-btn").at(0).simulate("click");
      await flushPromises();
      wrapper.update();

      expect(showErrorMessage).toHaveBeenCalledWith(
        "Error in updating bank detail, please try again!",
        "",
        true
      );
    });
  });

  it("should display error message, if sort code is invalid", async () => {
    props.bankDetail.sortCode = "123";
    const wrapper = mount(<BankDetail {...props} />);

    wrapper.find(".update-btn").last().simulate("click");
    wrapper.update();
    await flushPromises();

    expect(
      wrapper.find("p").contains("The sort code field must be valid.")
    ).toBeTruthy();
    expect(props.setCustomerDetails).not.toHaveBeenCalled();
  });

  it("should display error message, if account number is invalid", async () => {
    props.bankDetail.accountNumber = "74855";
    const wrapper = mount(<BankDetail {...props} />);

    wrapper.find(".update-btn").last().simulate("click");
    wrapper.update();
    await flushPromises();

    expect(
      wrapper
        .find("p")
        .contains("The account number field must be 8 digits long.")
    ).toBeTruthy();
    expect(props.setCustomerDetails).not.toHaveBeenCalled();
  });

  it("should not call updateBaseApplicationDetail api, if bank detail is not changed", async () => {
    mockUpdateBaseApplicationDetail.mockImplementationOnce(() =>
      Promise.resolve(undefined)
    );

    await act(async () => {
      const wrapper = mount(<BankDetail {...props} />);

      wrapper.find(".update-btn").at(0).simulate("click");
      await flushPromises();
      wrapper.update();

      expect(mockUpdateBaseApplicationDetail).toHaveBeenCalledTimes(0);
      expect(props.backToWizardStepper).toHaveBeenCalledTimes(0);
    });
  });
});
