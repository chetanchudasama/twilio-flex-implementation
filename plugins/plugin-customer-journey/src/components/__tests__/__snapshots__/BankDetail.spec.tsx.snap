// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BankDetail renders correctly 1`] = `
<div
  className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1 css-1a5ug3b"
>
  <div
    className="MuiGrid-container-29"
  >
    <div
      className="MuiCardContent-root-126 card-content"
    >
      <div
        className="MuiGrid-item-30 MuiGrid-grid-xs-8-65 bank-detail-header"
      >
        Bank details
      </div>
      <div
        className="MuiGrid-item-30 MuiGrid-grid-xs-8-65"
      >
        <div
          className="MuiFormControl-root-131 MuiFormControl-marginDense-133 MuiFormControl-fullWidth-134 CustomInputContent-formControl-127"
        >
          <label
            className="MuiFormLabel-root-146 MuiFormLabel-filled-150 MuiInputLabel-root-135 MuiInputLabel-formControl-140 MuiInputLabel-animated-143 MuiInputLabel-shrink-142 MuiInputLabel-marginDense-141 CustomInputContent-label-128"
            data-shrink={true}
          >
            Bank name
          </label>
          <div
            className="MuiInputBase-root-166 MuiOutlinedInput-root-153 MuiInputBase-formControl-167 MuiInputBase-marginDense-173 CustomInputContent-input-129 "
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="MuiPrivateNotchedOutline-root-183 MuiOutlinedInput-notchedOutline-160"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="MuiPrivateNotchedOutline-legend-184"
                style={
                  Object {
                    "width": 0,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <input
              aria-invalid={false}
              className="MuiInputBase-input-176 MuiOutlinedInput-input-161 MuiInputBase-inputMarginDense-177 MuiOutlinedInput-inputMarginDense-162"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value="Bank of England"
            />
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-item-30 MuiGrid-grid-xs-8-65"
      >
        <div
          className="MuiFormControl-root-131 MuiFormControl-marginDense-133 MuiFormControl-fullWidth-134 CustomInputContent-formControl-127"
        >
          <label
            className="MuiFormLabel-root-146 MuiFormLabel-filled-150 MuiInputLabel-root-135 MuiInputLabel-formControl-140 MuiInputLabel-animated-143 MuiInputLabel-shrink-142 MuiInputLabel-marginDense-141 CustomInputContent-label-128"
            data-shrink={true}
          >
            Months at bank
          </label>
          <div
            className="MuiInputBase-root-166 MuiOutlinedInput-root-153 MuiInputBase-formControl-167 MuiInputBase-marginDense-173 CustomInputContent-input-129 "
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="MuiPrivateNotchedOutline-root-183 MuiOutlinedInput-notchedOutline-160"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="MuiPrivateNotchedOutline-legend-184"
                style={
                  Object {
                    "width": 0,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <input
              aria-invalid={false}
              className="MuiInputBase-input-176 MuiOutlinedInput-input-161 MuiInputBase-inputMarginDense-177 MuiOutlinedInput-inputMarginDense-162"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value={11}
            />
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-item-30 MuiGrid-grid-xs-8-65"
      >
        <div
          className="MuiFormControl-root-131 MuiFormControl-marginDense-133 MuiFormControl-fullWidth-134 CustomInputContent-formControl-127"
        >
          <label
            className="MuiFormLabel-root-146 MuiFormLabel-filled-150 MuiInputLabel-root-135 MuiInputLabel-formControl-140 MuiInputLabel-animated-143 MuiInputLabel-shrink-142 MuiInputLabel-marginDense-141 CustomInputContent-label-128"
            data-shrink={true}
          >
            Name on bank account
          </label>
          <div
            className="MuiInputBase-root-166 MuiOutlinedInput-root-153 MuiInputBase-formControl-167 MuiInputBase-marginDense-173 CustomInputContent-input-129 "
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="MuiPrivateNotchedOutline-root-183 MuiOutlinedInput-notchedOutline-160"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="MuiPrivateNotchedOutline-legend-184"
                style={
                  Object {
                    "width": 0,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <input
              aria-invalid={false}
              className="MuiInputBase-input-176 MuiOutlinedInput-input-161 MuiInputBase-inputMarginDense-177 MuiOutlinedInput-inputMarginDense-162"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value="Edward Gibbs"
            />
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-item-30 MuiGrid-grid-xs-8-65"
      >
        <div
          className="MuiFormControl-root-131 MuiFormControl-marginDense-133 MuiFormControl-fullWidth-134 CustomInputContent-formControl-127"
        >
          <label
            className="MuiFormLabel-root-146 MuiFormLabel-filled-150 MuiInputLabel-root-135 MuiInputLabel-formControl-140 MuiInputLabel-animated-143 MuiInputLabel-shrink-142 MuiInputLabel-marginDense-141 CustomInputContent-label-128"
            data-shrink={true}
          >
            Account no.
          </label>
          <div
            className="MuiInputBase-root-166 MuiOutlinedInput-root-153 MuiInputBase-formControl-167 MuiInputBase-marginDense-173 CustomInputContent-input-129 "
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="MuiPrivateNotchedOutline-root-183 MuiOutlinedInput-notchedOutline-160"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="MuiPrivateNotchedOutline-legend-184"
                style={
                  Object {
                    "width": 0,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <input
              aria-invalid={false}
              className="MuiInputBase-input-176 MuiOutlinedInput-input-161 MuiInputBase-inputMarginDense-177 MuiOutlinedInput-inputMarginDense-162"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value="31510604"
            />
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-item-30 MuiGrid-grid-xs-8-65"
      >
        <div
          className="MuiFormControl-root-131 MuiFormControl-marginDense-133 MuiFormControl-fullWidth-134 CustomInputContent-formControl-127"
        >
          <label
            className="MuiFormLabel-root-146 MuiFormLabel-filled-150 MuiInputLabel-root-135 MuiInputLabel-formControl-140 MuiInputLabel-animated-143 MuiInputLabel-shrink-142 MuiInputLabel-marginDense-141 CustomInputContent-label-128"
            data-shrink={true}
          >
            Sort code
          </label>
          <div
            className="MuiInputBase-root-166 MuiOutlinedInput-root-153 MuiInputBase-formControl-167 MuiInputBase-marginDense-173 CustomInputContent-input-129 "
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="MuiPrivateNotchedOutline-root-183 MuiOutlinedInput-notchedOutline-160"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="MuiPrivateNotchedOutline-legend-184"
                style={
                  Object {
                    "width": 0,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <input
              aria-invalid={false}
              className="MuiInputBase-input-176 MuiOutlinedInput-input-161 MuiInputBase-inputMarginDense-177 MuiOutlinedInput-inputMarginDense-162"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value="56-57-77"
            />
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-item-30 MuiGrid-grid-xs-8-65 monthly-income-container"
      >
        <div
          className="MuiFormControl-root-131 MuiFormControl-marginDense-133 MuiFormControl-fullWidth-134 CustomInputContent-formControl-127"
        >
          <label
            className="MuiFormLabel-root-146 MuiFormLabel-filled-150 MuiFormLabel-required-151 MuiInputLabel-required-139 MuiInputLabel-root-135 MuiInputLabel-formControl-140 MuiInputLabel-animated-143 MuiInputLabel-shrink-142 MuiInputLabel-marginDense-141 CustomInputContent-label-128"
            data-shrink={true}
          >
            Net monthly income
            <span
              className="MuiFormLabel-asterisk-152"
            >
               *
            </span>
          </label>
          <div
            className="MuiInputBase-root-166 MuiOutlinedInput-root-153 MuiInputBase-formControl-167 MuiInputBase-marginDense-173 CustomInputContent-input-129 "
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="MuiPrivateNotchedOutline-root-183 MuiOutlinedInput-notchedOutline-160"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="MuiPrivateNotchedOutline-legend-184"
                style={
                  Object {
                    "width": 0,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <input
              aria-invalid={false}
              className="MuiInputBase-input-176 MuiOutlinedInput-input-161 MuiInputBase-inputMarginDense-177 MuiOutlinedInput-inputMarginDense-162"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value={30000}
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiCardActions-root-185 card-action"
    >
      <div
        className="MuiGrid-item-30 MuiGrid-grid-xs-8-65 MuiCardActions-action-187"
      >
        <button
          className="MuiButtonBase-root-214 MuiButton-root-188 MuiButton-contained-199 MuiButton-containedSecondary-201 MuiButton-raised-202 MuiButton-raisedSecondary-204 update-btn"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-189"
          >
            Update Details
          </span>
          <span
            className="MuiTouchRipple-root-217"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
